appium/options/android/common/adb/adb_options.py

```diff
@@ -0,0 +1,102 @@
+# Licensed to the Software Freedom Conservancy (SFC) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The SFC licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from datetime import timedelta
+from typing import Any
+
+from appium.options.common.supports_capabilities import SupportsCapabilities
+
+
+class AdbOptionsDescriptor:
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def __get__(self, obj: Any, cls: Any) -> Any:
```

Comment:
looking forward to see proper type declarations on these args. getattr does not look right in this context

Reply:
@mykola-mokhnach thanks for taking time to review my PR. I will work on your review comments. Thanks!

Reply:
> 

```
capability = TypeVar("capability", bound=SupportsCapabilities)

class OptionsDescriptor:
    def __init__(self, name: str) -> None:
        self.name = name

    def __get__(self, obj: capability, cls: type[capability]) -> Any:
        return getattr(obj, "get_capability")(self.name)

    def __set__(self, obj: capability, value: Any) -> capability:
        getattr(obj, "set_capability")(self.name, value)
```
@mykola-mokhnach are the above type hints okay?